viadot/utils.py

```diff
@@ -345,3 +351,31 @@ def union_dict(*dicts):
 
     """
     return dict(chain.from_iterable(dct.items() for dct in dicts))
+
+
+def handle_if_empty_file(if_empty: Literal["warn", "skip", "fail"] = "warn"):
+    if if_empty == "warn":
+        logger.warning("The input file is empty.")
+    elif if_empty == "skip":
+        raise SKIP()
+    elif if_empty == "fail":
+        raise ValueError("The input file is empty.")
+
+
+def check_if_empty_file(
+    path: str,
+    if_empty: Literal["warn", "skip", "fail"] = "warn",
+    file_extension: Literal[".parquet", ".csv"] = None,
+    file_sep: str = "\t",
+):
+    if os.stat(path).st_size == 0:
+        handle_if_empty_file(if_empty)
+
+    elif file_extension == ".parquet":
+        df = pd.read_parquet(path)
+        if "_viadot_downloaded_at_utc" in df.columns and len(df.columns) == 1:
+            handle_if_empty_file(if_empty)
+    elif file_extension == ".csv":
```

Comment:
Should include a comment why you're doing this (why not use `os.stat` and why use `_viadot_downloaded_at_utc` column)